### PR TITLE
add more aggressive legacy toggle behavior and better tests

### PR DESCRIPTION
**Pull request by czue**
hopefully addresses http://manage.dimagi.com/default.asp?181197

@dannyroberts @snopoke

### CODE REVIEW

File: corehq/ex-submodules/casexml/apps/phone/exceptions.py [hunk 1]
```diff
@@ -58,3 +58,6 @@ class InvalidDomainError(OwnershipCleanlinessError):
 
 class InvalidOwnerIdError(OwnershipCleanlinessError):
     pass
+
+class SimplifiedSyncAssertionError(Exception):
```
Comment (dimagimon) [inline corehq/ex-submodules/casexml/apps/phone/exceptions.py]: E302 expected 2 blank lines, found 1

File: corehq/ex-submodules/casexml/apps/phone/tests/test_new_sync.py [hunk 2]
```diff
@@ -167,3 +169,60 @@ def test_new_then_old_sync(self):
...
+        cls.user_id = uuid.uuid4().hex
+        cls.user = User(user_id=cls.user_id, username=uuid.uuid4().hex,
+                        password="changeme", date_joined=datetime(2014, 6, 6))
+
+    def test_legacy_support_toggle(self):
+        restore_config = RestoreConfig(self.project, user=self.user)
+        factory = CaseFactory(domain=self.project.name, case_defaults={'owner_id': self.user_id})
+        # create a parent and child case (with index) from one user
+        parent_id, child_id = [uuid.uuid4().hex for i in range(2)]
+        parent_id, child_id = ['parent', 'child']
```
Comment (snopoke) [inline corehq/ex-submodules/casexml/apps/phone/tests/test_new_sync.py]: This line overwrites the last

File: corehq/ex-submodules/casexml/apps/phone/tests/test_new_sync.py [hunk 3]
```diff
@@ -167,3 +169,60 @@ def test_new_then_old_sync(self):
...
+        with self.assertRaises(SimplifiedSyncAssertionError):
+            factory.create_or_update_cases([
+                CaseStructure(case_id=child_id, attrs={'owner_id': 'different'}),
+                CaseStructure(case_id=parent_id, attrs={'owner_id': 'different'}),
+            ], form_extras={'last_sync_token': sync_log._id})
+
+        # enabling the toggle should prevent the failure the second time
+        # though we also need to hackily set the request object in the threadlocals
+        LEGACY_SYNC_SUPPORT.set(self.domain, True, namespace='domain')
+        request = JsonObject(domain=self.domain)
```
Comment (snopoke) [inline corehq/ex-submodules/casexml/apps/phone/tests/test_new_sync.py]: Nice

File: corehq/ex-submodules/casexml/apps/phone/tests/test_new_sync.py (hunk 2 above)
Comment (czue) [inline corehq/ex-submodules/casexml/apps/phone/tests/test_new_sync.py]: whoops that was just for testing locally https://github.com/dimagi/commcare-hq/pull/8174/

**Pull request merged**